### PR TITLE
Add tutorial for changefeed to msk serverless

### PR DESCRIPTION
**Pull request by kathancox**
Fixes DOC-10965, DOC-10984

This PR adds a tutorial to the docs for changefeed running to the MSK Serverless.

Also adds the note re MSK Serverless not supporting auto topic creation.

Note: A follow-up to this PR will be to add instructions for MSK standard.

## Preview

[Tutorial page](https://deploy-preview-18903--cockroachdb-docs.netlify.app/docs/v24.2/stream-a-changefeed-to-amazon-msk-serverless.html)

### CODE REVIEW

File: src/current/v24.2/changefeed-sinks.md [hunk 1]
```diff
@@ -104,6 +104,8 @@ You can either manually create a topic in your Kafka cluster before starting the
 
 {{site.data.alerts.callout_info}}
 You must have the Kafka cluster setting [`auto.create.topics.enable`](https://kafka.apache.org/documentation/#brokerconfigs_auto.create.topics.enable) set to `true` for automatic topic creation. This will create the topic when the changefeed sends its first message. If you create the consumer before that, you will also need the Kafka consumer configuration [`allow.auto.create.topics`](https://kafka.apache.org/documentation/#consumerconfigs_allow.auto.create.topics) to be set to `true`.
+
+For [Amazon MSK Serverless cluster](#amazon-msk), you must create topics manually. Refer to the [Stream a Changefeed to Amazon MSK Serverless]({% link {{ page.version.version }}/stream-a-changefeed-to-amazon-msk-serverless.md %}) tutorial for an example.
```
Comment (andyyang890) [inline src/current/v24.2/changefeed-sinks.md:108]: nit: s/cluster/clusters/

File: src/current/v24.2/changefeed-sinks.md [hunk 2]
```diff
@@ -227,12 +229,14 @@ Changefeeds can deliver messages to Amazon MSK clusters ([Amazon Managed Streami
 
 {% include_cached new-in.html version="v24.2" %} Changefeeds can deliver messages to MSK and MSK Serverless clusters using AWS IAM roles.
 
-{% comment %}will change out the links here to CRDB tutorials in a follow-up PR.{% endcomment %}
+For initial setup guides, refer to:
 
-For initial setup guides, refer to the AWS documentation:
+- [MSK clusters](https://aws.amazon.com/blogs/apn/how-to-unlock-real-time-data-streams-with-cockroachdb-and-amazon-msk/)
```
Comment (andyyang890) [inline src/current/v24.2/changefeed-sinks.md:234]: Should we specify "MSK Provisioned" here?

**Review by andyyang890 — APPROVED** [review]
LGTM! I didn't run through the steps manually but they seem similar to what I did when I did the performance testing.

Adding @asg0451 as a reviewer in case he wanted to take a look too.

File: src/current/v24.2/stream-a-changefeed-to-amazon-msk-serverless.md [hunk 3]
```diff
@@ -0,0 +1,296 @@
...
+    ~~~
+
+1. Enable the `kv.rangefeed.enabled` [cluster setting]({% link {{ page.version.version }}/cluster-settings.md %}):
+
+    {% include_cached copy-clipboard.html %}
+    ~~~ sql
+    SET CLUSTER SETTING kv.rangefeed.enabled = true;
+    ~~~
+
+1. Create an [external connection]({% link {{ page.version.version }}/create-external-connection.md %}) for the MSK Serverless URI. External connections define a name for an external connection while passing the provider URI and query parameters:
```
Comment (asg0451) [inline src/current/v24.2/stream-a-changefeed-to-amazon-msk-serverless.md:229]: consider adding a note that you can also just specify the sink uri in the feed directly, without having to go through an external connection.

**Review by asg0451 — APPROVED** [review]

File: src/current/v24.2/changefeed-sinks.md (hunk 2 above)
Comment (kathancox) [reply src/current/v24.2/changefeed-sinks.md:234]: I understand why, but Amazon call the "provisioned" just Amazon MSK, so I was trying to keep to that where possible.

File: src/current/v24.2/changefeed-sinks.md (hunk 1 above)
Comment (kathancox) [reply src/current/v24.2/changefeed-sinks.md]: Done!

File: src/current/v24.2/stream-a-changefeed-to-amazon-msk-serverless.md [hunk 4]
```diff
@@ -0,0 +1,302 @@
...
+
+    Once you've updated the policy, click **Update policy**.
+
+## Step 4. Connect the client to the MSK Serverless cluster
+
+In this step, you'll prepare the client to connect to the MSK Serverless cluster, create a Kafka topic, and consume messages that the changefeed sends.
+
+1. Ensure that your client can connect to the MSK Serverless cluster. This tutorial uses an EC2 instance running Kafka as the client. Navigate to the summary page for the client EC2 instance. Click on the **Actions** dropdown. Click **Security**, and then select **Modify IAM role**.
+1. On the **Modify IAM role** page, select the role you created for the MSK Serverless cluster (`msk-serverless-role`) that contains the policy created in [Step 2](#step-2-create-an-iam-policy-and-role-to-access-the-msk-serverless-cluster). Click **Update IAM role**.
+1. Open a terminal and connect to your Kafka client. Check that your `client.properties` file contains the correct SASL and secuity configuration, like the following:
```
Comment (kathancox) [inline src/current/v24.2/stream-a-changefeed-to-amazon-msk-serverless.md:148]:
```suggestion
1. Open a terminal and connect to your Kafka client. Check that your `client.properties` file contains the correct SASL and security configuration, like the following:
```

File: src/current/v24.3/stream-a-changefeed-to-amazon-msk-serverless.md [hunk 5]
```diff
@@ -0,0 +1,302 @@
+---
+title: Stream a Changefeed to Amazon MSK Serverless
+summary: Learn how to connect a changefeed to stream data to an Amazon MSK Serverless cluster.
+toc: true
+---
+
+{% include_cached new-in.html version="v24.2" %} [Changefeeds]({% link {{ page.version.version }}/change-data-capture-overview.md %}) can stream change data to [Amazon MSK Serverless clusters](https://docs.aws.amazon.com/msk/latest/developerguide/serverless.html) (Amazon Managed Streaming for Apache Kafka), which is an Amazon MSK cluster type that automatically scales your capacity.
```
Comment (kathancox) [inline src/current/v24.3/stream-a-changefeed-to-amazon-msk-serverless.md:7]:
```suggestion
[Changefeeds]({% link {{ page.version.version }}/change-data-capture-overview.md %}) can stream change data to [Amazon MSK Serverless clusters](https://docs.aws.amazon.com/msk/latest/developerguide/serverless.html) (Amazon Managed Streaming for Apache Kafka), which is an Amazon MSK cluster type that automatically scales your capacity.
```

File: src/current/v24.2/stream-a-changefeed-to-amazon-msk-serverless.md [hunk 6]
```diff
@@ -0,0 +1,302 @@
...
+toc: true
+---
+
+{% include_cached new-in.html version="v24.2" %} [Changefeeds]({% link {{ page.version.version }}/change-data-capture-overview.md %}) can stream change data to [Amazon MSK Serverless clusters](https://docs.aws.amazon.com/msk/latest/developerguide/serverless.html) (Amazon Managed Streaming for Apache Kafka), which is an Amazon MSK cluster type that automatically scales your capacity.
+
+MSK Serverless requires [IAM authentication](https://docs.aws.amazon.com/IAM/latest/UserGuide/introduction.html?icmpid=docs_iam_console) for the changefeed to connect to the cluster.
+
+In this tutorial, you'll set up an MSK Serverless cluster and connect a changefeed with IAM authentication. You'll create the MSK Serverless cluster with an IAM policy and role. CockroachDB and a Kafka client will assume the IAM role in order to connect to the MSK Serverless cluster. Then you'll set up the Kafka client to consume the changefeed messages and start the changefeed on the CockroachDB cluster.
+
+An overview of the workflow involves:
```
Comment (rmloveland) [inline src/current/v24.2/stream-a-changefeed-to-amazon-msk-serverless.md:13]: i feel like this list kind of says the previous paragraph again and also duplicates the TOC

could prolly be deleted IMO but that's just, like, an opinion

File: src/current/v24.2/stream-a-changefeed-to-amazon-msk-serverless.md [hunk 7]
```diff
@@ -0,0 +1,302 @@
...
+    SET CLUSTER SETTING kv.rangefeed.enabled = true;
+    ~~~
+
+1. To connect the changefeed to the MSK Serverless cluster, the URI must contain the following parameters:
+    - The MSK Serverless cluster endpoint prefixed with the `kafka://` scheme, for example: `kafka://boot-vab1abab.c1.kafka-serverless.us-east-1.amazonaws.com:9098`.
+    - `tls_enabled` set to `true`.
+    - `sasl_enabled` set to `true`.
+    - `sasl_mechanism` set to `AWS_MSK_IAM`.
+    - `sasl_aws_region` set to the region of the MSK Serverless cluster.
+    - `sasl_aws_iam_role_arn` set to the ARN for the IAM role (`msk-serverless-role`) that has the permissions outlined in [Step 2.2](#step-2-create-an-iam-policy-and-role-to-access-the-msk-serverless-cluster).
```
Comment (rmloveland) [inline src/current/v24.2/stream-a-changefeed-to-amazon-msk-serverless.md:235]: 'step 2.2' is a nice touch

**Review by rmloveland — APPROVED** [review]
LGTM!  (AFAICT the big MSK tutorial content is the same for v24.2 and v24.3 so I carefully read v24.2 and carefully skimmed v24.3 😄 )

File: src/current/v24.2/stream-a-changefeed-to-amazon-msk-serverless.md [hunk 8]
```diff
@@ -0,0 +1,294 @@
+---
+title: Stream a Changefeed to Amazon MSK Serverless
+summary: Learn how to connect a changefeed to stream data to an Amazon MSK Serverless cluster.
+toc: true
+---
+
+[Changefeeds]({% link {{ page.version.version }}/change-data-capture-overview.md %}) can stream change data to [Amazon MSK Serverless clusters](https://docs.aws.amazon.com/msk/latest/developerguide/serverless.html) (Amazon Managed Streaming for Apache Kafka), which is an Amazon MSK cluster type that automatically scales your capacity.
```
Comment (kathancox) [inline src/current/v24.2/stream-a-changefeed-to-amazon-msk-serverless.md:7]:
```suggestion
{% include_cached new-in.html version="v24.2" %} [Changefeeds]({% link {{ page.version.version }}/change-data-capture-overview.md %}) can stream change data to [Amazon MSK Serverless clusters](https://docs.aws.amazon.com/msk/latest/developerguide/serverless.html) (Amazon Managed Streaming for Apache Kafka), which is an Amazon MSK cluster type that automatically scales your capacity.
```

**Pull request merged**